server/routes/archiveGoal/archiveGoalFormValidator.test.ts

```diff
@@ -0,0 +1,84 @@
+import type { ArchiveGoalForm } from 'forms'
+import aValidArchiveGoalForm from '../../testsupport/archiveGoalFormTestDataBuilder'
+import validateArchiveGoalForm from './archiveGoalFormValidator'
+import ReasonToArchiveGoalValue from '../../enums/ReasonToArchiveGoalValue'
+
+describe('archiveGoalFormValidator', () => {
+  it('should validate given no errors', () => {
+    // Given
+    const form: ArchiveGoalForm = aValidArchiveGoalForm()
+
+    // When
+    const errors = validateArchiveGoalForm(form)
+
+    // Then
+    expect(errors).toEqual([])
+  })
+
+  it('should require a reason to be selected', () => {
+    // Given
+    const form: ArchiveGoalForm = {
+      reference: '95b18362-fe56-4234-9ad2-11ef98b974a3',
+      title: 'Learn French',
+    }
+
+    // When
+    const errors = validateArchiveGoalForm(form)
+
+    // Then
+    expect(errors).toEqual([{ href: '#reason', text: 'You must select a reason to archive the goal' }])
+  })
+
+  it.each([undefined, ''])('should require details when selecting OTHER', anInvalidReasonOther => {
+    // Given
+    const form: ArchiveGoalForm = {
+      reference: '95b18362-fe56-4234-9ad2-11ef98b974a3',
+      title: 'Learn French',
+      reason: ReasonToArchiveGoalValue.OTHER,
+      reasonOther: anInvalidReasonOther,
+    }
+
+    // When
+    const errors = validateArchiveGoalForm(form)
+
+    // Then
+    expect(errors).toEqual([{ href: '#reasonOther', text: 'Enter the reason you are archiving the goal' }])
+  })
+
+  it('other reason should be 200 chars or less', () => {
+    // Given
+    const anInvalidReasonOther = new Array(202).join('x')
+    const form: ArchiveGoalForm = {
+      reference: '95b18362-fe56-4234-9ad2-11ef98b974a3',
+      title: 'Learn French',
+      reason: ReasonToArchiveGoalValue.OTHER,
+      reasonOther: anInvalidReasonOther,
+    }
+
+    // When
+    const errors = validateArchiveGoalForm(form)
+
+    // Then
+    expect(errors).toEqual([
+      { href: '#reasonOther', text: 'The reason is limited to 200 characters but you entered 201' },
+    ])
+  })
+
+  it('other reason should be able to be 200 chars', () => {
+    // Given
+    const aValidReasonOther = new Array(201).join('x')
+    const form: ArchiveGoalForm = {
+      reference: '95b18362-fe56-4234-9ad2-11ef98b974a3',
+      title: 'Learn French',
+      reason: ReasonToArchiveGoalValue.OTHER,
+      reasonOther: aValidReasonOther,
+    }
+
+    // When
+    const errors = validateArchiveGoalForm(form)
+
+    // Then
+    expect(aValidReasonOther).toHaveLength(200)
```

Comment:
Similar comment to above in that I dont understand where we are apparently loosing a character from the length:
```
const aValidReasonOther = new Array(201).join('x')
```
and
```
expect(aValidReasonOther).toHaveLength(200)
```

But also I don't understand why we are making this assertion (on the length of the OTHER field) - we know how long it is (201) because we set it up on line 69.
There's nothing in the implementation that changes this value is there? 😕

Reply:
No I just left that assertion there because it was so weird 😂  also replaced with padEnd()